pepper-apis/src/main/java/org/broadinstitute/ddp/model/event/activityinstancecreation/ActivityInstanceCreatorFromAnswers.java

```diff
@@ -0,0 +1,165 @@
+import org.broadinstitute.ddp.db.dto.ActivityDto;
+import org.broadinstitute.ddp.db.dto.QuestionDto;
+import org.broadinstitute.ddp.exception.DDPException;
+import org.broadinstitute.ddp.model.activity.instance.answer.Answer;
+import org.broadinstitute.ddp.model.activity.instance.answer.CompositeAnswer;
+import org.broadinstitute.ddp.model.activity.instance.answer.PicklistAnswer;
+import org.broadinstitute.ddp.model.event.EventSignal;
+import org.broadinstitute.ddp.service.ActivityInstanceCreationService;
+import org.jdbi.v3.core.Handle;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Create  activity instances from the selected answers of parent instances.
+ * Question stable_id which answers to check is specified by parameter {@link #sourceQuestionStableId}.
+ *
+ * <p><b>Algorithm:</b>
+ * <ul>
+ *     <li>find answer (picklist) by 'sourceQuestionStableId' and detect selected options;</li>
+ *     <li>for each of selected options:</li>
+ *     <li> - create a new instance (of activity with ID='studyActivityId';</li>
+ *     <li> - if `targetQuestionStableId` is specified then copy source answers to target answers;</li>
+ * </ul>
+ */
+public class ActivityInstanceCreatorFromAnswers extends ActivityInstanceCreatorDefault {
+
+    private static final Logger LOG = LoggerFactory.getLogger(ActivityInstanceCreatorFromAnswers.class);
+
+    private final String sourceQuestionStableId;
+    private final String targetQuestionStableId;
+
+    public ActivityInstanceCreatorFromAnswers(
+            long studyActivityId,
+            String sourceQuestionStableId,
+            String targetQuestionStableId,
+            ActivityInstanceCreationService creationService) {
+        super(studyActivityId, creationService);
+        this.sourceQuestionStableId = sourceQuestionStableId;
+        this.targetQuestionStableId = targetQuestionStableId;
+    }
+
+    /**
+     * Create N activity instances.
+     * where N = number of answers in composite question `sourceQuestionStableId`.
+     * IF N > not_used_activities_count THEN N = not_used_activities_count,
+     * where not_used_activities_count = max_allowed_activity_instances - created_activity_instances.
+     */
+    @Override
+    public void create(Handle handle, EventSignal signal) {
+        JdbiActivityInstance jdbiActivityInstance = handle.attach(JdbiActivityInstance.class);
+        JdbiActivityInstanceStatus jdbiActivityInstanceStatus = handle.attach(JdbiActivityInstanceStatus.class);
+        JdbiActivity jdbiActivity = handle.attach(JdbiActivity.class);
+
+        ActivityDto activityDto = jdbiActivity.queryActivityById(studyActivityId);
+        creationService.checkSignalIfNestedTargetActivity(activityDto);
+
+        // detect parent instance id
+        Long parentInstanceId = creationService.detectParentInstanceId(activityDto);
+        if (parentInstanceId == null) {
+            throw new DDPException("Failed to detect parent instance ID");
```

Comment:
Wait, why are we requiring that there is a parent instance? We're not creating child activities of a parent activity. They're totally separate activities. We should get the "source" activity instance from the `signal` without requiring that there is a parent activity.

Reply:
Initially I thought that it detects a current activity. Now I see that I was wrong.. Fixed